clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12487,6 +12487,7 @@ def err_hlsl_pointers_unsupported : Error<
   "%select{pointers|references}0 are unsupported in HLSL">;
 def err_hlsl_missing_resource_class : Error<"HLSL resource needs to have [[hlsl::resource_class()]] attribute">;
 def err_hlsl_attribute_needs_intangible_type: Error<"attribute %0 can be used only on HLSL intangible type %1">;
+def err_hlsl_builtin_requires_resource: Error<"operand must be of __hlsl_resource_t type">;
```

Comment:
FYI, I am adding the same error in my PR: https://github.com/llvm/llvm-project/pull/114148

Reply:
Good to know. I really need to review that PR, sorry!

Reply:
No worries! I know you've been busy.